lifecycle-operator/apis/lifecycle/v1alpha4/keptnworkloadcersion_types_test.go

```diff
@@ -0,0 +1,448 @@
+
+	workload.SetCurrentPhase(common.PhaseAppDeployment.LongName)
+	require.Equal(t, common.PhaseAppDeployment.LongName, workload.GetCurrentPhase())
+
+	workload.Status.EndTime = v1.Time{Time: time.Time{}}
+	workload.Complete()
+	require.True(t, !workload.GetEndTime().IsZero())
+
+	require.Equal(t, "version", workload.GetVersion())
+
+	require.Equal(t, "trace1.workloadname.version.phase", workload.GetSpanKey("phase"))
+
+	retries := int32(5)
+	task := workload.GenerateTask(v1alpha3.KeptnTaskDefinition{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "task-def",
+			Labels: map[string]string{
+				"label1": "label2",
+			},
+			Annotations: map[string]string{
+				"annotation1": "annotation2",
+			},
+		},
+		Spec: v1alpha3.KeptnTaskDefinitionSpec{
+			Timeout: v1.Duration{
+				Duration: 5 * time.Second,
+			},
+			Retries: &retries,
+		},
+	}, common.PostDeploymentCheckType)
+	require.Equal(t, v1alpha3.KeptnTaskSpec{
+		Context: v1alpha3.TaskContext{
+			AppName:         workload.GetAppName(),
+			WorkloadVersion: workload.GetVersion(),
+			WorkloadName:    workload.GetParentName(),
+			TaskType:        string(common.PostDeploymentCheckType),
+			ObjectType:      "Workload",
+		},
+		TaskDefinition:   "task-def",
+		Parameters:       v1alpha3.TaskParameters{},
+		SecureParameters: v1alpha3.SecureParameters{},
+		Type:             common.PostDeploymentCheckType,
+		Timeout: v1.Duration{
+			Duration: 5 * time.Second,
+		},
+		Retries: &retries,
+	}, task.Spec)
+
+	require.Equal(t, map[string]string{
+		"label1": "label2",
+	}, task.Labels)
+
+	require.Equal(t, map[string]string{
+		"annotation1": "annotation2",
+	}, task.Annotations)
+
+	evaluation := workload.GenerateEvaluation(v1alpha3.KeptnEvaluationDefinition{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "eval-def",
+		},
+	}, common.PostDeploymentCheckType)
+	require.Equal(t, v1alpha3.KeptnEvaluationSpec{
+		AppName:              workload.GetAppName(),
+		WorkloadVersion:      workload.GetVersion(),
+		Workload:             workload.GetParentName(),
+		EvaluationDefinition: "eval-def",
+		Type:                 common.PostDeploymentCheckType,
+		RetryInterval: metav1.Duration{
+			Duration: 5 * time.Second,
+		},
+	}, evaluation.Spec)
+
+	require.Equal(t, "workload", workload.GetSpanName(""))
+
+	require.Equal(t, "workloadname/phase", workload.GetSpanName("phase"))
+
+	require.Equal(t, []attribute.KeyValue{
+		common.AppName.String("appname"),
+		common.WorkloadName.String("workloadname"),
+		common.WorkloadVersion.String("version"),
+		common.WorkloadNamespace.String("namespace"),
+	}, workload.GetSpanAttributes())
+
+	require.Equal(t, map[string]string{
+		"appName":              "appname",
+		"workloadName":         "workloadname",
+		"workloadVersion":      "version",
+		"workloadInstanceName": "workload",
+	}, workload.GetEventAnnotations())
+}
+
+//nolint:dupl
+func TestKeptnWorkloadVersion_DeprecateRemainingPhases(t *testing.T) {
+	workloadInstance := KeptnWorkloadVersion{
```

Comment:
let's rename this little guy as well 
```suggestion
	workloadVersion := KeptnWorkloadVersion{
```